core/listener.go

```diff
@@ -2,18 +2,23 @@ package core
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"time"
 
 	pubsub "github.com/libp2p/go-libp2p-pubsub"
 	"github.com/tendermint/tendermint/types"
 
 	libhead "github.com/celestiaorg/go-header"
 
 	"github.com/celestiaorg/celestia-node/header"
+	"github.com/celestiaorg/celestia-node/nodebuilder/p2p"
 	"github.com/celestiaorg/celestia-node/share/eds"
 	"github.com/celestiaorg/celestia-node/share/p2p/shrexsub"
 )
 
+const defaultListenTimeout = p2p.BlockTime * 2
```

Comment:
Mmm, we should not depend on node builder and isntead we should pass this value to the syncer